contracts/ronin/fast-finality/FastFinalityTracking.sol

```diff
@@ -0,0 +1,69 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.9;
+
+import "@openzeppelin/contracts/proxy/utils/Initializable.sol";
+import { IRoninValidatorSet } from "../../interfaces/validator/IRoninValidatorSet.sol";
+import { IFastFinalityTracking } from "../..//interfaces/IFastFinalityTracking.sol";
+import "../../extensions/collections/HasContracts.sol";
+import "../../utils/CommonErrors.sol";
+
+contract FastFinalityTracking is IFastFinalityTracking, Initializable, HasContracts {
+  /// @dev Mapping from epoch number => consensus address => number of QC vote
+  mapping(uint256 => mapping(address => uint256)) internal _tracker;
+  /// @dev The latest block that tracked the QC vote
+  uint256 internal _latestTrackingBlock;
+
+  modifier oncePerBlock() {
+    if (block.number <= _latestTrackingBlock) {
+      revert ErrOncePerBlock();
+    }
+
+    _latestTrackingBlock = block.number;
+    _;
+  }
```

Comment:
I can see the same pattern in this contract, we can reuse the error code?
https://github.com/axieinfinity/ronin-dpos-contracts/blob/75df9c7f6575630e3479f890a5bae22d0814bb7d/contracts/ronin/slash-indicator/SlashUnavailability.sol#L42-L49

Reply:
No need to refactor for now